app/src/main/java/com/hedvig/app/feature/offer/binders/PerilAdapter.kt

```diff
@@ -0,0 +1,47 @@
+package com.hedvig.app.feature.offer.binders
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.hedvig.android.owldroid.graphql.OfferPreSaleQuery
+import com.hedvig.app.BuildConfig
+import com.hedvig.app.R
+import com.hedvig.app.util.extensions.isDarkThemeActive
+import kotlinx.android.synthetic.main.offer_peril_item.view.*
+
+class PerilAdapter : RecyclerView.Adapter<PerilAdapter.ViewHolder>() {
+
+    var list: List<OfferPreSaleQuery.Peril> = listOf()
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ViewHolder(parent)
+
+    override fun getItemCount() = list.size
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.itemView.apply {
```

Comment:
Let's allow the `ViewHolder` to own all the binding code. This entails to setting up derived properties on the `ViewHolder` (`private val title = itemView.title`) and a method `fun bind(data: OfferPreSaleQuery.Peril)` which is the only thing called in the `onBindViewHolder`-method